c7n/resources/backup.py

```diff
@@ -87,3 +91,57 @@ def get_resources(self, resource_ids, cache=True):
 class KmsFilter(KmsRelatedFilter):
 
     RelatedIdsExpression = 'EncryptionKeyArn'
+
+
+@BackupVault.action_registry.register('modify-policy')
+class ModifyPolicyStatement(ModifyPolicyBase):
+    """Action to modify Backup Vault IAM policy statements.
+
+    :example:
+
+    .. code-block:: yaml
+
+           policies:
+              - name: backup-vault-get-policy
+                resource: backup-vault
+                filters:
+                  - type: value
+                    key: BackupVaultName
+                    value: "backup_vault_name"
+                actions:
+                  - type: modify-policy
+                    add-statements:
+                      - "Sid": "AddMe"
+                        "Effect": "Allow"
+                        "Principal": {"AWS": "arn:aws:iam::116249476610:root"}
+                        "Action": ["backup:GetBackupVaultAccessPolicy"]
+                        "Resource": "*"
+                    remove-statements: []
+    """
+    permissions = ('backup:PutBackupVaultAccessPolicy', 'backup:GetBackupVaultAccessPolicy')
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('backup')
+
+        for r in resources:
+            policy = json.loads(
+                client.get_backup_vault_access_policy(
```

Comment:
if a vault doesn't have an access policy associated, this will error out with a resource not found exception, in this context  it should instead catch that.